surmise/calibration.py

```diff
@@ -259,11 +265,37 @@ def predict(self, x=None, args=None):
         if 'predict' in dir(self.method):
             self.method.predict(info, self.info, self.emu, x, args)
         else:
-            emupred = self.emu.predict(x, self.theta.rnd(1000))
+            nsamp = 1000
+            emupred = self.emu.predict(x, self.theta.rnd(nsamp))
             info['mean'] = np.mean(emupred.mean(), 1)
             info['var'] = np.var(emupred.mean(), 1)
             info['rnd'] = (emupred.mean()).T
-        return prediction(info, self)
+
+        predobj = prediction(info, self)
+        predobj.empirical_coverage()
+        return predobj
+
+    @staticmethod
+    def cast_f64_dtype(x):
+        return np.array(x, dtype=np.float64)

```

Comment:
Refer to discussion about making this a package-wide helper function to avoid code duplication and to add as much type/sanity checking as possible in it.

Reply:
Resolved in e7c9af4190b4c5276d3d190277106762efbb2b6f